### PR TITLE
fix: also rename UserId to userId in method body

### PR DESCRIPTION
**Pull request by kesselb**
## Description

Follow-up for https://github.com/nextcloud-libraries/rector/pull/7

## Motivation and context

It seems reasonable to rename `UserId`  to `userId`  in the method body as well :rofl: 

Note: That only works for assignments. Need some more code to also replace it in function calls. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING.md** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/Rector/RenameUserIdRector.php [hunk 1]
```diff
@@ -74,6 +77,20 @@ public function refactor(Node $node): ?Node
...
+            foreach ($stmts as $stmt) {
+                if ($stmt instanceof Expression) {
+                    if ($stmt->expr instanceof Assign) {
+                        if ($stmt->expr->expr instanceof Variable && $stmt->expr->expr->name === 'UserId') {
+                            $stmt->expr->expr->name = 'userId';
+                        }
+                    }
+                }
+            }
+        }
```
Comment (ChristophWurst) [inline src/Rector/RenameUserIdRector.php:92]: How about early returns and early continues?

**Review by ChristophWurst — APPROVED** [review]

**Comment by come-nc** [comment]
Can you make it generic so that we use it for AppName and WebRoot as well?
ref: https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_26.html#dependency-injection-parameters

**Comment by kesselb** [comment]
> Can you make it generic so that we use it for AppName and WebRoot as well?

Good idea!

- Renamed to RenameParameterRector
- Added configuration objects
- Also replace the variables in method and static calls

**Comment by kesselb** [comment]
I have another follow-up in the queue, better detection of extended classes with reflection resolver.
However, psalm doesn't allow me to push the change because the constructor does not initialize all properties :see_no_evil: 

image

**Review by come-nc — APPROVED** [review]

**Comment by come-nc** [comment]
> I have another follow-up in the queue, better detection of extended classes with reflection resolver. However, psalm doesn't allow me to push the change because the constructor does not initialize all properties 🙈
> 
> image

Initialize them, or make them nullable and init to null?

**Comment by come-nc** [comment]
> The base branch requires all commits to be signed. Learn more about signing commits.

@kesselb

**Comment by kesselb** [comment]
Done ;)

Note: The current version can not replace in if/foreach/etc.

**Comment by come-nc** [comment]
> Done ;)
> 
> Note: The current version can not replace in if/foreach/etc. If you think the rector is too risky, then we should not enable it by default.

I’d say it’s fine, constructor are usually simple, and if it misses one CI should spot the problem quite easily.

**Pull request merged**